### DISCUSSION TITLE
citation key not produced (as specified/at all)

### DISCUSSION THREAD

**Discussion by jkurle** [retorquere/zotero-better-bibtex#2593] [Q&A] [answered]
Hello,

I don't think this is a bug, it is more likely that I am not using the Add-on correctly, so any help is greatly appreciated. I could not find my problem in the existing or recent issues.

I have Zotero v6.0.26 and only just installed the most recent version of Better BibTeX v6.7.94. I have not changed any Better BibTeX settings except for the citation key formula, which I specified as follows:

authEtal2.lower + "_" + year + type(preprint) + "_wp"

In my understanding, this should create citation keys such as angrist.krueger_1991 or angrist.krueger_1991_wp if it is of type preprint. This initially worked once as I started to populate my library.

However, now the citation keys are not created any more. New entries are instead given some random key (e.g. zotero-189). Even if I right click on the item > Better BibTeX > Refresh BibTeX key, the citation key remains this cryptic default key.

Any ideas why this is not working (anymore)? Is my setup or my workflow wrong?

**Comment by retorquere** [comment, marked as answer]
I'm not sure why that ever worked -- `type` aborts the formula it is part of entirely when the condition is not met, regardless of its position within the formula, and that behavior has always been that way. In this case I think you'll need

```type(preprint) + authEtal2.lower + "_" + year + "_wp" | authEtal2.lower + "_" + year```

**Reply by jkurle** [reply]
Thanks so much, that fixed it!

Not sure why it worked initially, I simply thought that `type` aborts the rest of the formula if it evaluates to false, and would therefore only add the `"_wp"` part if it evaluates to true.

**Reply by retorquere** [reply]
I'm still working on a way to make that possible, but I've been distracted by the zotero 7 port.
